x/proof/keeper/session.go

```diff
@@ -69,6 +72,56 @@ func (k msgServer) queryAndValidateSessionHeader(
 	return onChainSession, nil
 }
 
+// validateClaimWindow returns an error if the given session is not eligible for claiming.
+func (k msgServer) validateClaimWindow(
+	ctx context.Context,
+	sessionHeader *sessiontypes.SessionHeader,
+) error {
+	logger := k.Logger()
+
+	sharedParams := k.sharedKeeper.GetParams(ctx)
+
+	// Get the on-chain session end height for the given session header.
+	sessionEndHeight := shared.GetSessionEndHeight(&sharedParams, sessionHeader.GetSessionEndBlockHeight())
```

Comment:
I dont think we need to call `shared.GetSessionEndHeight()` since it will always be the same as `sessionHeader.GetSessionEndBlockHeight()`

Reply:
He's got a point.